src/Evolution/Systems/NewtonianEuler/Limiters/KxrcfTci.hpp

```diff
@@ -0,0 +1,201 @@
+  }
+  alg::for_each(neighbor_data, [&mesh](const auto& neighbor_and_data) noexcept {
+    if (UNLIKELY(neighbor_and_data.second.mesh != mesh)) {
+      ERROR("The Kxrcf TCI does not yet support p-refinement");
+      // Removing this limitation will require generalizing the surface
+      // integrals to make sure the meshes are consistent.
+    }
+  });
+
+  bool inflow_boundaries_present = false;
+  double inflow_area = 0.;
+  double inflow_delta_density = 0.;
+  double inflow_delta_energy = 0.;
+
+  for (const auto& neighbor_and_data : neighbor_data) {
+    const auto& neighbor = neighbor_and_data.first;
+    const auto& dir = neighbor.first;
+
+    // Skip computations on boundary if the boundary is external
+    // Note: may need to revisit this choice for problems that feed in shocks
+    // through the boundary condition. Such a change may also require changing
+    // the tags at the call site to ensure that ALL face normals are grabbed
+    // from the databox (and not only the internal face normals, as occurs
+    // when grabbing `Tags::Interface<Tags::InternalDirections, ...>`).
+    if (unnormalized_normals.find(dir) == unnormalized_normals.end()) {
+      continue;
+    }
+
+    const size_t sliced_dim = dir.dimension();
+    const size_t index_of_edge =
+        (dir.side() == Side::Lower ? 0 : mesh.extents()[sliced_dim] - 1);
+    const auto momentum_on_edge = data_on_slice(
+        cons_momentum_density, mesh.extents(), sliced_dim, index_of_edge);
+    const auto momentum_dot_normal =
+        dot_product(momentum_on_edge, unnormalized_normals.at(dir));
+
+    // Skip boundaries with no significant inflow
+    // Note: the cutoff value here is small but arbitrarily chosen.
+    if (min(get(momentum_dot_normal)) > -1e-12) {
+      continue;
+    }
+    inflow_boundaries_present = true;
+
+    // This mask has value 1. for momentum_dot_normal < 0.
+    //                     0. for momentum_dot_normal >= 0.
+    const DataVector inflow_mask = 1. - step_function(get(momentum_dot_normal));
+    // Mask is then weighted pointwise by the Jacobian determinant giving
+    // surface integrals in inertial coordinates. This Jacobian determinant is
+    // given by the product of the volume Jacobian determinant with the norm of
+    // the unnormalized face normals.
+    const DataVector weighted_inflow_mask =
+        inflow_mask *
+        get(data_on_slice(det_logical_to_inertial_jacobian, mesh.extents(),
+                          sliced_dim, index_of_edge)) *
+        get(magnitude(unnormalized_normals.at(dir)));
+
+    inflow_area +=
+        definite_integral(weighted_inflow_mask, mesh.slice_away(sliced_dim));
+
+    // This is the step that is incompatible with h/p refinement. For use with
+    // h/p refinement, would need to correctly obtain the neighbor solution on
+    // the local grid points.
+    const auto neighbor_vars_on_edge =
+        data_on_slice(neighbor_and_data.second.volume_data, mesh.extents(),
```

Comment:
you say "edge" here. does that mean this limiter only works in 2d? If not, maybe "surface" instead of edge? (or face)

Reply:
What do you think of renaming `*_on_edge` -> `*_on_slice`? This will be more consistent with the existing `data_on_slice` function, for example.

Reply:
I'll allow it :smile_cat: